src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,16 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> {
+                    if (i % 2 != 0) {
+                        return numbers.get(i) - 1;
+                    } else {
+                        return numbers.get(i);
+                    }
+                })
+                .filter(n -> n % 2 != 0)
+                .average().orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .average()
                .getAsDouble()
```
it will generate this exception by default if there is no value